pkg/client/record/event.go

```diff
@@ -42,49 +55,82 @@ type EventRecorder interface {
 // your binary. Subsequent calls will be ignored. The return value can be ignored
 // or used to stop recording, if desired.
 func StartRecording(recorder EventRecorder, source api.EventSource) watch.Interface {
+	// Set up our own personal buffer of events so that we can clear out GetEvents'
+	// broadcast channel as quickly as possible to avoid causing the relatively more
```

Comment:
The alternatives are to either get rid of the broadcast functionality that allows multiple subscribers or to mess with the watch.Broadcast class to allow for dropping rather than blocking. I'll do the latter during my free block of time later this afternoon unless I hear otherwise from you before then.